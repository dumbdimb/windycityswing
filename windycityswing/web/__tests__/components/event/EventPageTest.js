import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import EventPage from '..\\..\\..\\src\\scripts\\components\\event\\EventPage';

it('renders correctly', () => {
    expect(renderer.create(<EventPage match={{params: {eventName: 'fizz', date: '2017-07-03'}}}/>).toJSON()).toMatchSnapshot();
});

it('has the correct Google Calendar link', () => {
    const tree = shallow(<EventPage match={{params: {eventName: 'fizz', date: '2017-07-03'}}}/>);

    tree.setState({
        event: {
            "title": "Fizz",
            "className": "fizz",
            "siteTitle": "Fizz",
            "date": {
                "year": 2017,
                "month": 5,
                "day": 1,
                "weekday": 1
            },
            "timezone": {
                "tzid": "America/Chicago",
                "startTimestamp": "20170502T200000",
                "endTimestamp": "20170503T000000"
            },
            "recurrence": {
                "rule": "dayOfWeek",
                "dayOfWeek": 1,
                "exceptions": ["2017-05-29"]
            },
            "facebookUrl": "https://www.facebook.com/Fizz-Swing-Dancing-73706742234/",
            "organizer": "Fizz Staff",
            "instructor": "Fizz Staff",
            "schedule": [
                {
                    "hour": 20,
                    "minute": 0,
                    "event": "Swing lesson"
                },
                {
                    "hour": 21,
                    "minute": 0,
                    "event": "Dancing"
                },
                {
                    "hour": 0,
                    "minute": 0,
                    "event": "End"
                }
            ],
            "info": [
                "Occurs every Monday.",
                "All ages event."
            ],
            "infoUrl": "",
            "attendance": "50-100",
            "music": {
                "type": "DJ",
                "url": "",
                "by": "Fizz Staff"
            },
            "floorSize": "Medium",
            "location": {
                "addressName": "Lion Head Pub",
                "addressOne": "2251 N Lincoln Ave",
                "addressTwo": "2nd Floor",
                "city": "Chicago",
                "state": "IL",
                "zip": "60614"
            },
            "dressCode": "Casual",
            "danceStyles": [
                "East coast swing",
                "Lindy hop",
                "Charleston",
                "Balboa"
            ],
            "parking": "Street parking",
            "price": "$7/person for general public, $5/person for students",
            "colors": {
                "bg": "#654639",
                "text": "white"
            }
        }
    })

    const googleCalendarLink = tree.find('.event-details__link--gcal');

    expect(googleCalendarLink.props().href).toEqual(`https://calendar.google.com/calendar/render?action=TEMPLATE&text=Fizz&dates=20170704T010000Z/20170704T050000Z&details=Occurs every Monday.%0D%0AAll ages event.&location=Lion Head Pub, 2251 N Lincoln Ave 2nd Floor,+Chicago,+IL 60614#eventpage_6`);
});