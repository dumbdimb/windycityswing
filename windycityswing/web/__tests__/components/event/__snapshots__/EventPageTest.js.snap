// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="event-details"
>
  <span
    className="event-details__title"
  >
    Fizz
  </span>
  <div
    className="event-details__content"
  >
    <div
      className="event-details__text-content"
    >
      <span
        className="event-details__header"
      >
        LINKS
      </span>
      <div
        className="event-details__links"
      >
        <a
          className="event-details__link event-details__link--facebook"
          href="https://www.facebook.com/Fizz-Swing-Dancing-73706742234/"
          target="_blank"
        >
          <span
            className="event-details__link-text"
          >
            Facebook
          </span>
          <svg
            className="event-details__link-icon"
            fill="transparent"
            stroke="white"
            viewBox="0 0 16 16"
          >
            <path
              d="M8 3 L1 3 L1 15 L13 15 L13 8"
            />
            <path
              d="M10 1 L15 1 L15 6"
            />
            <line
              x1="7"
              x2="15"
              y1="9"
              y2="1"
            />
          </svg>
        </a>
        <a
          className="event-details__link event-details__link--gcal"
          href="https://calendar.google.com/calendar/render?action=TEMPLATE&text=Fizz&dates=20170704T010000Z/20170704T050000Z&details=Occurs every Monday.%0D%0AAll ages event.&location=Lion Head Pub, 2251 N Lincoln Ave 2nd Floor,+Chicago,+IL 60614#eventpage_6"
          target="_blank"
        >
          <span
            className="event-details__link-text"
          >
            Google Calendar
          </span>
          <svg
            className="event-details__link-icon"
            fill="transparent"
            stroke="white"
            viewBox="0 0 16 16"
          >
            <path
              d="M8 3 L1 3 L1 15 L13 15 L13 8"
            />
            <path
              d="M10 1 L15 1 L15 6"
            />
            <line
              x1="7"
              x2="15"
              y1="9"
              y2="1"
            />
          </svg>
        </a>
      </div>
      <span
        className="event-details__header"
      >
        DESCRIPTION
      </span>
      <span
        className="event-details__description"
      >
        Occurs every Monday.
      </span>
      <span
        className="event-details__description"
      >
        All ages event.
      </span>
      <span
        className="event-details__header"
      >
        LOCATION
      </span>
      <span
        className="event-details__address-name"
      >
        Lion Head Pub
      </span>
      <span
        className="event-details__address-street"
      >
        2251 N Lincoln Ave, 2nd Floor
      </span>
      <span
        className="event-details__address-city"
      >
        Chicago, IL 60614
      </span>
      <span
        className="event-details__header"
      >
        PRICE
      </span>
      <span
        className="event-details__price"
      >
        $7/person for general public, $5/person for students
      </span>
      <span
        className="event-details__header"
      >
        SCHEDULE
      </span>
      <div
        className="event-details__date--container"
      >
        <span
          className="event-details__date"
        >
          Monday, July 3rd
        </span>
        <span
          className="event-details__date--relative"
        >
          
        </span>
      </div>
      <table>
        <tbody>
          <tr>
            <td
              className="event-details__schedule__time"
            >
              8:00 PM
            </td>
            <td
              className="event-details__schedule__event"
            >
              Swing lesson
            </td>
          </tr>
          <tr>
            <td />
            <td
              className="event-details__schedule__duration event-details__schedule__duration-one"
            >
              1 hr
            </td>
          </tr>
          <tr>
            <td
              className="event-details__schedule__time"
            >
              9:00 PM
            </td>
            <td
              className="event-details__schedule__event"
            >
              Dancing
            </td>
          </tr>
          <tr>
            <td />
            <td
              className="event-details__schedule__duration event-details__schedule__duration-three"
            >
              3 hr
            </td>
          </tr>
          <tr>
            <td
              className="event-details__schedule__time"
            >
              Midnight
            </td>
            <td
              className="event-details__schedule__event"
            >
              End
            </td>
          </tr>
        </tbody>
      </table>
      <span
        className="event-details__header"
      >
        DRESS CODE
      </span>
      <span
        className="event-details__dress-code"
      >
        Casual
      </span>
    </div>
    <div
      className="event-details__map-container"
    >
      <span
        className="event-details__header"
      >
        MAP
      </span>
      <iframe
        className="event-details__map"
        src="https://maps.google.com/maps?q=Lion%20Head%20Pub%2C%202251%20N%20Lincoln%20Ave%2C%20Chicago%2C%20IL&amp;t=&amp;z=14&amp;ie=UTF8&amp;iwloc=&output=embed"
      />
    </div>
  </div>
</div>
`;
